### ISSUE TITLE
Write tests for status of edX application

### ISSUE THREAD

**Issue by blarghmatey** [mitodl/salt-ops#248]
In order to ensure that our edX instances are healthy we need to be able to run tests to verify the state of the system both at creation and on a periodic schedule so that we can be notified of soft-fail situations.

**Comment by blarghmatey** [comment]
Some things to test:
- [ ] edxapp supervisor process is running
- [ ] edxapp listen port is accessible
- [ ] edxapp instance can reach services including:
  - [ ] RabbitMQ
  - [ ] Elasticsearch
  - [ ] MySQL
  - [ ] Elastic File Store (NFS)
- [ ] edxapp can reach forum service
- [ ] edxapp can reach xqueue endpoint

**Comment by shaidar** [comment]
This is mainly complete. We tried going beyond a simple socket connection to test connectivity to several of the dependent services, however the socket module in testinfra does not support that. We submitted a [PR](https://github.com/philpep/testinfra/pull/190) to the testinfra library, however our use cases is somewhat different and PR will not be merged. For enhanced testing, we might need to revisit [goss](https://github.com/aelsabbahy/goss) and use that going forward.
